### DISCUSSION TITLE
Design a container (objects) to ensure that all types in this container are unique

### DISCUSSION THREAD

**Discussion by Arc-huangjingtong** [dotnet/csharplang#8373] [Q&A] [answered]
How to design a container (objects) to ensure that all types in this container are unique
Avoid using reflection, including methods such as GetType, typeof

**Comment by lindexi** [comment, marked as answer]
I do not think this is belong here. It is the [Dependency injection](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines) issues, which define in https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyInjection/README.md

So that I think you should create the discussion to [dotnet/runtime](https://github.com/dotnet/runtime)

**Reply by Arc-huangjingtong** [reply]
thanks
